tidb-server/main.go

```diff
@@ -391,6 +433,7 @@ func overrideConfig(cfg *config.Config) {
 		p, err = strconv.Atoi(*port)
 		terror.MustNil(err)
 		cfg.Port = uint(p)
+		cfg.UpdateTempStoragePath()
```

Comment:
If `tidb-server` starts without command-line argument, `UpdateTempStoragePath` may never be called.